src/LegacyEventMapper.php

```diff
@@ -19,6 +19,73 @@
 
 class LegacyEventMapper
 {
+    // Maps background/legacy event types to their equivalent CloudEvent types.
+    // For more info on event mappings see
+    // https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/docs/mapping.md
+    private $ceTypeMap = [
+        'google.pubsub.topic.publish' => 'google.cloud.pubsub.topic.v1.messagePublished',
+        'providers/cloud.pubsub/eventTypes/topic.publish' => 'google.cloud.pubsub.topic.v1.messagePublished',
+        'google.storage.object.finalize' => 'google.cloud.storage.object.v1.finalized',
+        'google.storage.object.delete' => 'google.cloud.storage.object.v1.deleted',
+        'google.storage.object.archive' => 'google.cloud.storage.object.v1.archived',
+        'google.storage.object.metadataUpdate' => 'google.cloud.storage.object.v1.metadataUpdated',
+        'providers/cloud.firestore/eventTypes/document.write' => 'google.cloud.firestore.document.v1.written',
+        'providers/cloud.firestore/eventTypes/document.create' => 'google.cloud.firestore.document.v1.created',
+        'providers/cloud.firestore/eventTypes/document.update' => 'google.cloud.firestore.document.v1.updated',
+        'providers/cloud.firestore/eventTypes/document.delete' => 'google.cloud.firestore.document.v1.deleted',
+        'providers/firebase.auth/eventTypes/user.create' => 'google.firebase.auth.user.v1.created',
+        'providers/firebase.auth/eventTypes/user.delete' => 'google.firebase.auth.user.v1.deleted',
+        'providers/google.firebase.analytics/eventTypes/event.log' => 'google.firebase.analytics.log.v1.written',
+        'providers/google.firebase.database/eventTypes/ref.create' => 'google.firebase.database.document.v1.created',
+        'providers/google.firebase.database/eventTypes/ref.write' => 'google.firebase.database.document.v1.written',
+        'providers/google.firebase.database/eventTypes/ref.update' => 'google.firebase.database.document.v1.updated',
+        'providers/google.firebase.database/eventTypes/ref.delete' => 'google.firebase.database.document.v1.deleted',
+        'providers/cloud.storage/eventTypes/object.change' => 'google.cloud.storage.object.v1.finalized',
+    ];
+
+    // CloudEvent service names.
+    private $firebaseAuthCEService = 'firebaseauth.googleapis.com';
+    private $firebaseCEService = 'firebase.googleapis.com';
+    private $firebaseDBCEService = 'firebasedatabase.googleapis.com';
+    private $firestoreCEService = 'firestore.googleapis.com';
+    private $pubSubCEService = 'pubsub.googleapis.com';
+    private $storageCEService = 'storage.googleapis.com';
```

Comment:
I suggest making these private constants:

```php
    private const FIREBASE_AUTH_CE_SERVICE = 'firebaseauth.googleapis.com';
    private const FIREBASE_CE_SERVICE = 'firebase.googleapis.com';
    private const FIREBSE_DB_SERVICE = 'firebasedatabase.googleapis.com';
    private const FIRESTORE_CE_SERVICE = 'firestore.googleapis.com';
    private const PUBSUB_CE_SERVICE = 'pubsub.googleapis.com';
    private const STORAGE_CE_SERVICE = 'storage.googleapis.com';
```

And then `$ceServiceMap` and `$ceResourceRegexMap` can be defined as part of the class (and also static):

```php
        private static $ceServiceMap = [
            'providers/cloud.firestore/' => self::FIRESTORE_CE_SERVICE,
            'providers/google.firebase.analytics/' => self::FIREBASE_CE_SERVICE,
            'providers/firebase.auth/' => self::FIREBASE_AUTH_CE_SERVICE,
            'providers/google.firebase.database/' => self::FIREBASE_DB_CE_SERVICE,
            'providers/cloud.pubsub/' => self::PUBSUB_CE_SERVICE,
            'providers/cloud.storage/' => self::STORAGE_CE_SERVICE
        ];

        private static $ceResourceRegexMap = [
            self::FIREBASE_CE_SERVICE => '#^(projects/[^/]+)/(events/[^/]+)$#',
            self::FIREBASE_DB_CE_SERVICE => '#^(projects/_/instances/[^/]+)/(refs/.+)$#',
            self::FIRESTORE_CE_SERVICE => '#^(projects/[^/]+/databases/\(default\))/(documents/.+)$#',
            self::STORAGE_CE_SERVICE => '#^(projects/_/buckets/[^/]+)/(objects/.+)$#',
        ];
```

Reply:
Done.